### DISCUSSION TITLE
RLS insert problem even if i'm logged in

### DISCUSSION THREAD

**Discussion by Arthurhrd** [supabase/supabase#11248] [Questions]
Hi,
I have an issue with my policies on supabase, i try to insert some articles into my tables.
I already have the select policy for all users, but when i add the policy for authenticated users only, i can't insert, this error appear :
```
new row violates row-level security policy for table "articles"
```

My insert function :
```
const { data, error } = await supabase
        .from('articles')
        .insert([
            { author_id:user.id,title:title,contents: contents }
        ],{ returning: "minimal" })
```

My policy :
```sql
CREATE POLICY "policy_name"
ON public.articles
FOR INSERT 
TO authenticated 
WITH CHECK (true);
```

Thanks for your response

**Comment by GaryAustin1** [comment]
This almost certainly means you are not signed in when you make the call.   The other cause is not meeting select, but you don't have .select() and are returning minimal (v1?).

You can look at the network tab in dev console, or the API Edge Log on the Dashboard to see what your headers are.  Your authenticated header should have a jwt much longer than the anon key (in the apikey header) if logged in.

You could also switch your TO in the policy to anon to confirm you are not logged in.

**Reply by GaryAustin1** [reply]
The important key is your `authenticated` header.  It should have 'bearer longjwt'  If that header is the same as the apikey header then you are not logged in when you make the call.   This could be your session has been cleared, or you are making these calls server side and don't have the correct setup.

**Reply by Arthurhrd** [reply]
Ok so yes the apikey and the authorization key are the same and the same as the anon key.
But how can i get the information from the user if i'm not logged in ?

**Reply by GaryAustin1** [reply]
Hard to know as you have not talked about your signin process or environment.  

One example is on signUp, it returns user but not session from the call if you have email confirmation on.

**Comment by Arthurhrd** [comment]
In the header of the request i got an apikey which is the same as the anon key.

and yes i had returning minimal  because i read some error about

But i think i'm well logged in because i can display with :
```
          <pre><code>{JSON.stringify(user, null, 2)}</code></pre>
```
all the information about the user logged

**Reply by james-william-r** [reply]
Did you ever figure this out? It's difficult to say without seeing an example of your code, but I suspect that the session isn't being applied correctly to the Supabase client. This could be for one of many reasons though, so if you're still stuck, share the full code of the view that your working with and I'll see if I can help! ☺️
